src/Sparrow/Json/BlittableJsonReaderObject.cs

```diff
@@ -628,15 +636,19 @@ private int StringValidation(int stringOffset)
                         PropertiesValidation(tokenType, propOffsetSize, propIdSize, propValueOffset, numberOfPropsNames);
                         break;
                     case BlittableJsonToken.Integer:
-                        ReadVariableSizeIntInReverse(_mem, propValueOffset, out offset);
+                        ReadVariableSizeLong(propValueOffset);
                         break;
                     case BlittableJsonToken.Float:
                         var floatLen = ReadNumber(_mem + objStartOffset - propOffset, 1);
+                        var pointFlag = true;
                         for (var j = 0; j < floatLen; j++)
                         {
-                            var floatDigit = ReadNumber((_mem + objStartOffset - propOffset + i), sizeof(byte));
-                            if (floatDigit == '.')
+                            var floatDigit = ReadNumber((_mem + objStartOffset - propOffset + 1 + j), sizeof(byte));
+                            if ((floatDigit == '.') && (pointFlag))
```

Comment:
Will still allow `02002022.2222.222222222.222222`
Will also allow for numbers that are too big.